pkg/operator/controller/externaldns/deployment.go

```diff
@@ -598,3 +602,78 @@ func buildStringMapHash(data map[string]string) (string, error) {
 	}
 	return buildMapHash(m)
 }
+
+func securityContextChanged(current, updated, desired *corev1.SecurityContext) (bool, *corev1.SecurityContext) {
+	desiredCopy := desired.DeepCopy()
```

Comment:
This won't be needed if `updated` variable is set along the way, without returning.